ghost/core/core/server/services/email-service/BatchSendingService.js

```diff
@@ -262,14 +273,51 @@ class BatchSendingService {
                     .select('members.id', 'members.uuid', 'members.email', 'members.name').limit(BATCH_SIZE + 1);
 
                 if (members.length > 0) {
-                    totalCount += Math.min(members.length, BATCH_SIZE);
-                    const batch = await this.retryDb(
-                        async () => {
-                            return await this.createBatch(email, segment, members.slice(0, BATCH_SIZE));
-                        },
-                        {...this.#getBeforeRetryConfig(email), description: `createBatch email ${email.id} segment ${segment}`}
-                    );
-                    batches.push(batch);
+                    // Determine how many members to include in this batch
+                    const remainingCustomDomainCapacity = domainWarmupLimit - totalCount;
+                    const membersToProcess = Math.min(members.length, BATCH_SIZE);
+
+                    // Calculate batch splits
+                    const batchesToCreate = [];
+
+                    if (remainingCustomDomainCapacity > 0 && remainingCustomDomainCapacity < membersToProcess) {
```

Comment:
I think this would do better as an explanatory variable 🤔  This is hard to follow otherwise. Maybe something like `shouldUseSplitBatch`?